tests/semantic_analyser.cpp

```diff
@@ -5686,4 +5695,27 @@ TEST_F(SemanticAnalyserTest, typeinfo_if_constexpr)
       R"(kprobe:f { $x = "xyz"; if (typeinfo($x) == typeinfo("abc")) { $x = "foo"; } else { $x = 2; } })");
 }
 
+TEST_F(SemanticAnalyserTest, typevalid_expr)
+{
+  // We can use `valid` as a regular expression.
+  test(R"(kprobe:f { $x = 1; print(typevalid($x)); })");
+  test(R"(kprobe:f { $x = 1; print(typevalid($y)); })");
+
+  // We get regular errors from `typevalid`.
+  test(R"(kprobe:f { $x = 1; print(typevalid($x, $y)); })", Error{ R"(
+stdin:1:20-39: ERROR: syntax error, unexpected ",", expecting )
+kprobe:f { $x = 1; print(typevalid($x, $y)); }
+                   ~~~~~~~~~~~~~~~~~~~
+)" });
+}
+
+TEST_F(SemanticAnalyserTest, typevalid_if_constexpr)
+{
+  // We will prune `typevalid`.
+  test(
```

Comment:
As mentioned previously, it would be nice to add a lot more tests but it's not a blocker.